src/main/kotlin/org/rust/ide/newProject/ui/RsNewProjectPanel.kt

```diff
@@ -5,52 +5,147 @@
     }
 
-    val data: ConfigurationData get() = ConfigurationData(rustProjectSettings.data, templateButtons[0].isSelected)
+    private var isCargoGenerateInstalled = false
+    private val downloadCargoGenerateLink = Link("Install cargo-generate using Cargo", action = {
+        val cargoExecutable = rustProjectSettings.data.toolchain?.rawCargo()?.cargoExecutable
+        if (cargoExecutable != null) {
+            object : Task.Backgroundable(null, "Installing cargo-generate") {
+                override fun shouldStartInBackground(): Boolean = false
+                override fun onSuccess() = update()
+
+                override fun run(indicator: ProgressIndicator) {
+                    indicator.isIndeterminate = true
+                    GeneralCommandLine(cargoExecutable)
+                        .withParameters(listOf("install", "cargo-generate"))
+                        .execute(this@RsNewProjectPanel, listener = object : ProcessListener {
+                            override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                                indicator.text = event.text.trim()
+                            }
+
+                            override fun processTerminated(event: ProcessEvent) {
+                                if (event.exitCode == 0) return
+
+                                PopupUtil.showBalloonForComponent(
+                                    templatePane,
+                                    "Failed to install cargo-generate",
+                                    MessageType.ERROR,
+                                    true,
+                                    this@RsNewProjectPanel
+                                )
+                            }
+
+                            override fun startNotified(event: ProcessEvent) {}
+
+                        })
+                }
+            }.queue()
+        }
+    }).apply { isVisible = false }
+
+    private val updateDebouncer = UiDebouncer(this)
+
+    val data: ConfigurationData get() = ConfigurationData(rustProjectSettings.data, templateList.selectedValue)
 
     fun attachTo(layout: RsLayoutBuilder) = with(layout) {
         rustProjectSettings.attachTo(this)
-        if (showProjectTypeCheckbox) {
-            val buttonGroup = ButtonGroup()
-            templateButtons.forEachIndexed { index, button ->
-                buttonGroup.add(button)
-                row(if (index == 0) "Project template:" else "", button)
-            }
+
+        if (showProjectTypeSelection) {
+            component(JBLabel("Project template:"))
+            component(templatePane)
+            component(downloadCargoGenerateLink)
         }
+
+        update()
+    }
+
+    fun update() {
+        updateDebouncer.run(
+            onPooledThread = {
+                val cargo = rustProjectSettings.data.toolchain?.rawCargo()
+                cargo != null && !Cargo.checkNeedInstallCargoGenerate(cargo)
```

Comment:
Looks like we can check `cargo-generate` installation only in case of `RsCustomTemplate`